src/tcp_tunnel_tools.js

```diff
@@ -81,26 +69,19 @@ function createTunnel(proxyUrl, targetHost, providedOptions = {}, callback) {
 
 module.exports.createTunnel = createTunnel;
 
-function closeTunnel(serverPath, closeConnections, callback) {
-    const [hostname, port] = serverPath.split(':');
-    if (!hostname) throw new Error('serverPath must contain hostname');
-    if (!port) throw new Error('serverPath must contain port');
-
-    const promise = new Promise((resolve) => {
```

Comment:
I read in the docs that this returns a promise if the callback is not provided. I wonder if it makes sense to remove that capability and potentially break someone's code. What's the benefit?

Reply:
> I wonder if it makes sense to remove that capability and potentially break someone's code.

Indeed, this is breaking.

> What's the benefit?

There's no need for this code to be asynchronous. The `error` argument is either undefined or an Error if the server was already closed, which by the current usage here, is never the case.

Reply:
Yeah, it's true that the error argument is rather pointless. But someone might want to just plainly wait for the server to close before doing something else. Now they can't.